routes/drones.js

```diff
@@ -1,37 +1,61 @@
 const express = require('express');
-
+const droneModel = require('../models/drone.model');
+// console.log(droneModel)
 // require the Drone model here
 
 const router = express.Router();
 
-router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
+router.get('/drones/list', (req, res) => {
+  droneModel.find({})
+  .then((dronesFromDB) => {
+    res.render('drones/list', { dronesFromDB })
+  })
+  .catch((error) => `Error while fetching drones: ${error}`)
 });
 
 router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  res.render('drones/create-form')
 });
 
 router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  const { name, propellers, maxSpeed } = req.body;
+
+  droneModel.create({ name, propellers, maxSpeed })
+    .then(() => res.redirect("/drones/list"))
+    .catch(() => res.render("/drones/create"))
 });
 
-router.get('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+router.get('/drones/:id/update-form', (req, res) => {
+  const { id } = req.params
+
+  droneModel.findById(id)
+  .then((droneToEdit) => {
+    res.render("drones/update-form", droneToEdit)
+  })
+  .catch((error) =>
+    console.log(`Error while getting a drone to edit: ${error}`))
 });
 
-router.post('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+router.post('/drones/:id/update-form', (req, res) => {
+    const { id } = req.params;
+    const { name, propellers, maxSpeed } = req.body;
+
+    droneModel.findByIdAndUpdate(
+      id,
+      { name, propellers, maxSpeed },
+      { new: true }
+    )
+    .then(() => res.redirect('/drones/list'))
+    .catch(() => res.render("/drones/:id/update-form"))
```

Comment:
The first argument of res.render takes in a view. Is the string that is passed a reference to a view?